### ISSUE TITLE
Use logical shift right for BV '>>' operator

### ISSUE THREAD

**Issue by JPaja** [prove-rs/z3.rs#243]
I think operator `>>` should be implemented to use bvlshr as default since BitVectors dont have sign.

```rs
impl_binary_op!(
    BV<'ctx>,
    Shr,
    ShrAssign,
    shr,
    shr_assign,
    bvlshr,
    mk_const_bv
);
```

**Comment by waywardmonkeys** [comment]
@Pat-Lafon What do you think?

**Comment by Pat-Lafon** [comment]
The original comment might need a little bit more elaboration. The original pr #88 which added these operations specifically called out shr as the BV operation not to be added because of sign issues.

While BVs don't have a sign, the documentation I've read suggests that the theory relies on special operations that change whether the BV should be treated as having a sign.

If we are looking to have consistency with the python implementation, https://z3prover.github.io/api/html/classz3py_1_1_bit_vec_ref.html#a5977ca600992dbeda4a98b38c372133a shows `>>` as mapping to `bvashr` instead(the arithmetic version).

I'm not sure how surprising/unexpected this mapping is for users. For example https://stackoverflow.com/questions/13955385/z3py-do-a-roll#comment19249416_13955593 shows there is confusion... but if it makes sense for the python bindings then maybe not having `>>` implemented at all would be more confusing.

**Comment by toolCHAINZ** [comment]
I would agree we should not implement this (as much as I like implementing `std::ops` on things; it's nice!). The Rust `Shr` trait doesn't itself specify whether it is arithmetic or logical, and as far as I can tell, which one it is depends on the trait impl. I'd be hesitant to "just use one" as this seems likely to cause subtle bugs for users if they use the operator and thus don't think about whether it's performing the actual operation they wish to express.
